File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productlist/ProductAdapter.java
package com.gerigorila.agenticmigrationdemo.presentation.productlist;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.gerigorila.agenticmigrationdemo.R;
import com.gerigorila.agenticmigrationdemo.domain.model.Product;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class ProductAdapter extends RecyclerView.Adapter<ProductAdapter.ProductViewHolder> {

    private final List<Product> products = new ArrayList<>();
    private OnProductClickListener listener;

    public interface OnProductClickListener {
        void onProductClick(int productId);
    }

    public void setOnProductClickListener(OnProductClickListener listener) {
        this.listener = listener;
    }

    public void setProducts(List<Product> newProducts) {
        products.clear();
        products.addAll(newProducts);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ProductViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_product, parent, false);
        return new ProductViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProductViewHolder holder, int position) {
        Product product = products.get(position);
        holder.txtName.setText(product.getName());
        holder.txtPrice.setText(String.format(Locale.US, "$%.2f", product.getPrice()));
        holder.itemView.setOnClickListener(v -> {
            if (listener != null) {
                listener.onProductClick(product.getId());
            }
        });
    }

    @Override
    public int getItemCount() {
        return products.size();
    }

    static class ProductViewHolder extends RecyclerView.ViewHolder {
        final TextView txtName;
        final TextView txtPrice;

        ProductViewHolder(@NonNull View itemView) {
            super(itemView);
            txtName = itemView.findViewById(R.id.txt_name);
            txtPrice = itemView.findViewById(R.id.txt_price);
        }
    }
}

File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productlist/ProductAdapter.kt
package com.gerigorila.agenticmigrationdemo.presentation.productlist

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.gerigorila.agenticmigrationdemo.R
import com.gerigorila.agenticmigrationdemo.domain.model.Product
import java.util.Locale

class ProductAdapter : RecyclerView.Adapter<ProductAdapter.ProductViewHolder>() {

    private val products = mutableListOf<Product>()
    private var listener: ((Int) -> Unit)? = null

    fun setOnProductClickListener(listener: (Int) -> Unit) {
        this.listener = listener
    }

    fun setProducts(newProducts: List<Product>) {
        products.clear()
        products.addAll(newProducts)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProductViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_product, parent, false)
        return ProductViewHolder(view)
    }

    override fun onBindViewHolder(holder: ProductViewHolder, position: Int) {
        val product = products[position]
        holder.txtName.text = product.name
        holder.txtPrice.text = String.format(Locale.US, "$%.2f", product.price)
        holder.itemView.setOnClickListener {
            listener?.invoke(product.id)
        }
    }

    override fun getItemCount(): Int = products.size

    class ProductViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val txtName: TextView = itemView.findViewById(R.id.txt_name)
        val txtPrice: TextView = itemView.findViewById(R.id.txt_price)
    }
}
